File: app/src/main/java/fr/xgouchet/chronorg/ui/fragments/ProjectListFragment.java
package fr.xgouchet.chronorg.ui.fragments;

import android.content.Intent;
import android.support.annotation.NonNull;

import com.deezer.android.counsel.annotations.Trace;

import java.util.ArrayList;

import fr.xgouchet.chronorg.data.models.Project;
import fr.xgouchet.chronorg.ui.activities.ProjectDetailsActivity;
import fr.xgouchet.chronorg.ui.activities.ProjectEditActivity;
import fr.xgouchet.chronorg.ui.adapters.ProjectsAdapter;

/**
 * @author Xavier Gouchet
 */
@Trace
public class ProjectListFragment extends BaseListFragment<Project, ProjectsAdapter> {

    @Override protected ProjectsAdapter getAdapter() {
        return new ProjectsAdapter(new ArrayList<Project>(), this);
    }

    @Override public void showCreateItemUi() {
        // TODO handle tablet
        Intent intent = ProjectEditActivity.intentNewProject(getActivity());
        getActivity().startActivity(intent);
    }

    @Override public void showItem(@NonNull Project item) {
        Intent intent = ProjectDetailsActivity.buildIntent(getActivity(), item);

        getActivity().startActivity(intent);
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/ui/fragments/ProjectListFragment.kt
package fr.xgouchet.khronorg.ui.fragments

import android.os.Bundle
import android.view.View
import fr.xgouchet.khronorg.data.models.Project
import fr.xgouchet.khronorg.ui.adapters.BaseAdapter
import fr.xgouchet.khronorg.ui.adapters.ProjectAdapter

/**
 * @author Xavier F. Gouchet
 */
class ProjectListFragment : ListFragment<Project>() {


    override val adapter: BaseAdapter<Project> = ProjectAdapter(this)

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        fab.setOnClickListener { e -> presenter.itemSelected(null) }
    }
}